test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala

```diff
@@ -1603,3 +1628,26 @@ object ExportObjSetterNamed_= { // scalastyle:ignore
   @JSExport
   val x = 1
 }
+
+object TopLevelExports {
+  @JSExportTopLevel("org.scalajs.testsuite.jsinterop.toplevel.basic")
+  def basic(): Int = 1
+
+  @JSExportTopLevel("org.scalajs.testsuite.jsinterop.toplevel.overload")
+  def overload(x: String): String = "Hello " + x
+
+  @JSExportTopLevel("org.scalajs.testsuite.jsinterop.toplevel.overload")
+  def overload(x: Int, y: Int*): Int = x + y.sum
+
+  var myVar: Int = _
+
+  @JSExportTopLevel("org.scalajs.testsuite.jsinterop.toplevel.set")
+  def setMyVar(x: Int): Unit = myVar = x
+
+  object Nested {
+    var myVar: Int = _
+
+    @JSExportTopLevel("org.scalajs.testsuite.jsinterop.toplevel.setNested")
+    def setMyVar(x: Int): Unit = myVar = x
+  }
+}
```

Comment:
Could you also add a top-level object that is not reachable by any other way than through its top-level exports? I'd like to get confidence that the analyzer does the right thing here.